src/components/explore/exploreList/ExploreList.tsx

```diff
@@ -0,0 +1,152 @@
+import { useCallback, useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import type { NextRouter } from 'next/router'
+import { List, ListItem, Typography, useTranslation } from '@okp4/ui'
+import type { DeepReadonly, UseState, UseTranslationResponse } from '@okp4/ui'
+import './exploreList.scss'
+import { fetchConfig, formatDate } from '../../../utils'
+import type { ExploreListLayout } from '../../../pages/dataverse/explore'
+import type { DataspaceDto } from '../../../dto/DataspaceDto'
+import type { DatasetDto } from '../../../dto/DatasetDto'
+import type { ServiceDto } from '../../../dto/ServiceDto'
+
+type ExploreListProps = {
+  readonly layout: ExploreListLayout
+  readonly range: number
+  readonly sortBy: string
+}
+
+type ItemDescriptionProps = {
+  readonly type: string
+  readonly categories: Array<string>
+}
+
+type ItemRightElementProps = {
+  readonly provider: string
+  readonly updatedOn: string
+}
+
+type DataverseEntity = DataspaceDto | DatasetDto | ServiceDto
+
+const ItemDescription = ({ type, categories }: DeepReadonly<ItemDescriptionProps>): JSX.Element => (
+  <div className="okp4-explore-list-item-description">
+    <Typography color="inverted-text" fontSize="small">
+      {type}
+    </Typography>
+    <Typography as="p" color="inverted-text" fontSize="small">
+      {categories.join(', ')}
+    </Typography>
+  </div>
+)
+
+const ItemRightElement = ({
+  provider,
+  updatedOn
+}: DeepReadonly<ItemRightElementProps>): JSX.Element => {
+  const { t, i18n }: UseTranslationResponse = useTranslation()
+
+  return (
+    <Typography color="inverted-text" fontSize="small">
+      {t('explore:by', {
+        provider: provider,
+        updated: formatDate(updatedOn, i18n.language)
+      })}
+    </Typography>
+  )
+}
+
+const fetchItems = async (
+  range: number,
+  sortBy: string
+): Promise<DeepReadonly<DataverseEntity[]>> => {
+  const config = await fetchConfig()
+  const responses = [
+    await fetch(`${config.app.apiUri}/dataverse/dataspace/`),
+    await fetch(`${config.app.apiUri}/dataverse/dataset/`),
+    await fetch(`${config.app.apiUri}/dataverse/service/`)
+  ]
+
+  const items = (
+    await Promise.all(
+      responses.map(
+        async (item: DeepReadonly<Response>): Promise<DataverseEntity> =>
+          item.ok ? await item.json() : []
+      )
+    )
+  ).flat()
```

Comment:
Not efficient. Use the `Promise.all` to request concurrently all the resources use `then` chaining  and `map` idioms wisely:

Roughly, just to give you the idea:

```typescript
const items = await Promise.all([
   fetch('xxx'.then( item => item.json() ),
   fetch('yyy'.then( item => item.json() ),
  ...
])
```

And you can wrapp the json retrieval into a local function if you want...